### PR TITLE
Support use of single quote in when statement

### PR DESCRIPTION
**Pull request by luckschulze**
This change allowed to use single or double quotes the block use When.

### CODE REVIEW

File: src/liquid/blocks/when.cr [hunk 1]
```diff
@@ -2,7 +2,7 @@ require "./block"
 
 module Liquid::Block
   class When < InlineBlock
-    SIMPLE_EXP = /^\s*when ((?:\s*"([^"]+)",?)+)\s*$/
+    SIMPLE_EXP = /^\s*when ((?:\s*['"]([^"]+)['"],?)+)\s*$/
```
Comment (hugopl) [inline src/liquid/blocks/when.cr:5]: This accepts things like `when 'potato"`.

File: spec/blocks_spec.cr [hunk 2]
```diff
@@ -69,13 +69,17 @@ describe Liquid do
         when_node = Block::When.new("when \"cookie\", \"biscuit\"")
         when_node << Block::Raw.new("This is a cookie")
         case_node << when_node
+        when_node = Block::When.new("when \'potato\'")
```
Comment (setanta) [inline spec/blocks_spec.cr:72]: You don't need to escape the single quotes inside the double quotes here.

**Review by hugopl — CHANGES_REQUESTED** [review]

File: spec/parser_spec.cr [hunk 3]
```diff
@@ -97,6 +98,9 @@ describe Parser do
     when_node = Block::When.new("when \"cookie\", \"biscuit\"")
     when_node << Block::Raw.new(" This is a cookie\n  ")
     case_node << when_node
+    when_node = Block::When.new("when \'potato\'")
```
Comment (setanta) [inline spec/parser_spec.cr:101]: Ditto here for removing the escape from the single quotes.

**Review by setanta — CHANGES_REQUESTED** [review]

File: src/liquid/blocks/when.cr (hunk 1 above)
Comment (setanta) [reply src/liquid/blocks/when.cr:5]: This and `when "potato'` are good negative test cases.

Comment (hugopl) [reply src/liquid/blocks/when.cr:5]: it also do not accept things like `when 'pota"to'`

File: spec/parser_spec.cr [hunk 4]
```diff
@@ -152,4 +156,10 @@ describe Parser do
 
     template.root.children.should eq expected
   end
+
+  it "should raise error if try to parse case statement using single and double quotes together" do
+    expect_raises(InvalidNode) do
+      when_node = Block::When.new("when 'cake\"")
+    end
```
Comment (setanta) [inline spec/parser_spec.cr:163]: Add the inverse
```suggestion
    end
    expect_raises(InvalidNode) do
      when_node = Block::When.new("when \"cake'")
    end
```

**Review by setanta — CHANGES_REQUESTED** [review]

File: src/liquid/blocks/when.cr [hunk 5]
```diff
@@ -12,7 +12,7 @@ module Liquid::Block
 
     def initialize(content : String)
       if match = content.match(SIMPLE_EXP)
-        @when_values = match[1].split(",").map { |value| /.*"(.*)".*/.match(value).not_nil![1] }
+        @when_values = match[2].gsub("\"", "").split(",").map { |value| value.strip }
```
Comment (hugopl) [inline src/liquid/blocks/when.cr:15]: To avoid some extra memory allocations you could use `.split(/\s*,\s*/)`.

**Review by hugopl — CHANGES_REQUESTED** [review]

**Review by hugopl — APPROVED** [review]

**Review by setanta — APPROVED** [review]

**Pull request merged**